deployment-scanner/src/main/java/org/jboss/as/server/deployment/scanner/FileSystemDeploymentService.java

```diff
@@ -780,6 +804,44 @@ private boolean handleCompositeResult(ModelNode resultNode, StringBuilder failur
     }
 
     /**
+     * Checks that given directory if readable & writable and prints a warning if the check fails. Warning is only
+     * printed once and is not repeated until the condition is fixed and broken again.
+     *
+     * @param directory deployment directory
+     * @return does given directory exist and is readable and writable?
+     */
+    private boolean checkDeploymentDir(File directory) {
+        if (!directory.exists()) {
+            if (deploymentDirAccessible) {
+                deploymentDirAccessible = false;
+                ROOT_LOGGER.directoryIsNonexistent(deploymentDir.getAbsolutePath());
+            }
+        }
+        else if (!directory.isDirectory()) {
+            if (deploymentDirAccessible) {
+                deploymentDirAccessible = false;
+                ROOT_LOGGER.isNotADirectory(deploymentDir.getAbsolutePath());
+            }
+        }
+        else if (!directory.canRead()) {
+            if (deploymentDirAccessible) {
+                deploymentDirAccessible = false;
+                ROOT_LOGGER.directoryIsNotReadable(deploymentDir.getAbsolutePath());
+            }
+        }
+        else if (!directory.canWrite()) {
+            if (deploymentDirAccessible) {
+                deploymentDirAccessible = false;
+                ROOT_LOGGER.directoryIsNotWritable(deploymentDir.getAbsolutePath());
```

Comment:
There needs to be a parameter or something to disable the logging from this method if the information has already been logged. Otherwise it looks like the same logging will happen 3 times at boot (bootTimeScan, startScanner, scan) and then periodically thereafter every time the scanner runs.